futures-core/src/future/mod.rs

```diff
@@ -8,6 +8,11 @@ pub use core::future::Future;
 mod future_obj;
 pub use self::future_obj::{FutureObj, LocalFutureObj, UnsafeFutureObj};
 
+#[cfg(feature = "alloc")]
+/// An owned dynamically typed [`Future`] for use in cases where you can't
+/// statically type your result or need to add some indirection.
+pub type BoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + Send + 'a>>;
```

Comment:
I'd also like to see non-`Send` versions of these-- maybe `BoxFutureLocal`? unsure-- we can sort that out in a separate change :)

Reply:
(similarly, adding a `boxed_local` would be great)